linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -6920,7 +6880,6 @@ private[optimizer] object OptimizerCore {
       case ApplyStatically(_, receiver, _, _, Nil) => isTrivialArg(receiver)
       case ApplyStatic(_, _, _, Nil)               => true
 
-      case ArrayLength(array)        => isTrivialArg(array)
       case ArraySelect(array, index) => isTrivialArg(array) && isTrivialArg(index)
 
       case AsInstanceOf(inner, _) => isSimpleArg(inner)
```

Comment:
Arguably we might not want it for `WrapAsThrowable` and `UnwrapAsThrowable`, but those won't actually happen in practice. Even then, at call-site there's a good chance that they can be simplified with their expansion, so it might even be better anyway.

For the other ones, IMO we do want that. One I have seen in the diffs are `throw`s. If a method does nothing but throw a simple arg, inlining it is probably beneficial.

I did not change this in this round, but I remain open to changing it if you have a stronger opinion.